js/qmwp-search-correlations.js

```diff
@@ -30,111 +31,185 @@ quantimodoSearch.config(function ($httpProvider) {
+                        }
+                        $scope.loadData();
+                    }
+                );
+
+            };
+
+            $scope.vote = function (correlationSet, likeValue) {
+                console.log('Liked: ' + likeValue);
+                console.log(correlationSet);
+
+                var modalInstance = $uibModal.open({
+                    templateUrl: qmwpPluginUrl + '/templates/search-correlations/vote-confirm-modal.html',
+                    controller: 'voteModalInstanceController',
+                    resolve: {
+                        confirmationOptions: function () {
+                            return {
+                                correlation: correlationSet.originalCorrelation,
+                                likeValue: likeValue
                             }
-                        });
+                        }
                     }
-                    $scope.resultTitle = 'Strongly predicted by ' + variable;
-                    if ($scope.selectOutputAsType === 'effect') {
-                        $scope.resultTitle = 'Strongest predictors of ' + variable;
+                });
+
+                modalInstance.result.then(function () {
+                    //confirmed
+                    var prevVoted = correlationsVoteHelper.getPreviouslyVoted(correlationSet.originalCorrelation);
+
+                    if (prevVoted === likeValue) {
+                        likeValue = 'null';
                     }
-                    $scope.countAndTime = $scope.totalCorrelations.length +
-                        ' results  (' + (((new Date()).getTime() - timeToSearch) / 1000) + ' seconds)';
-                    if ($scope.totalCorrelations.length === 0) {
-                        $scope.resultTitle = 'Your search for variable ' + variable + ' does not have any results';
+
+                    QuantimodoSearchService.vote(correlationSet.originalCorrelation, likeValue, function (resp) {
+                        console.log(resp);
+
+                        correlationsVoteHelper.saveVotedCorrelation(correlationSet.originalCorrelation, likeValue);
+
+                        refreshCorrelationsConsiderVotes();
+
+                    })
+
+                }, function () {
+                    console.debug('dismissed');
+                });
+
+            };
+
+            if (QuantimodoSearchConstants.predefinedVariable && QuantimodoSearchConstants.predefinedVariableAs) {
+                console.log('Variable: ' + QuantimodoSearchConstants.predefinedVariable);
+                console.log('Variable as: ' + QuantimodoSearchConstants.predefinedVariableAs);
+
+                $scope.selectOutputAsType = QuantimodoSearchConstants.predefinedVariableAs;
+                $scope.searchVariable = QuantimodoSearchConstants.predefinedVariable;
+                $scope.showCorrelations($scope.searchVariable);
+            }
+
+            function refreshCorrelationsConsiderVotes() {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['refreshCorrelationsConsiderVotes' is defined but never used.](https://www.codacy.com/app/m_3/QuantiModo-WordPress-Plugin/file/1187541654/issues/source?bid=2836188&fileBranchId=2920508#l177)